tests/Mono.Android-Tests/jni/reuse-threads.c

```diff
@@ -200,3 +206,67 @@ rt_invoke_callback_on_new_thread (CB cb)
 	return 0;
 }
 
+/* We return -2 for errors, because -1 is reserved for the pthreads PTHREAD_CANCELED special value, indicating that the
+ * thread was canceled. */
+static int
+_register_type_from_new_thread (void *data)
+{
+	RegisterFromThreadContext *context = (RegisterFromThreadContext*)data;
+
+	if (context == NULL) {
+		return -100;
+	}
+
+	JNIEnv *env              = _get_env ("_register_type_from_new_thread");
+	jclass ClassLoader_class = (*env)->FindClass (env, "java/lang/ClassLoader");
+	jmethodID loadClass      = (*env)->GetMethodID (env, ClassLoader_class, "loadClass", "(Ljava/lang/String;)Ljava/lang/Class;");
+	jobject loaded_class     = (*env)->CallObjectMethod (env, context->class_loader, loadClass, context->java_type_name);
+
+	if ((*env)->ExceptionOccurred (env) != NULL) {
+		(*env)->ExceptionClear (env);
+		__android_log_print (ANDROID_LOG_INFO, "XA/RuntimeTest", "FAILURE: class '%s' cannot be loaded, Java exception thrown!", context->java_type_name);
+		return -101;
+	}
+
+	jmethodID Object_ctor    = (*env)->GetMethodID (env, loaded_class, "<init>", "()V");
+	jobject   instance       = (*env)->NewObject (env, loaded_class, Object_ctor);
```

Comment:
you should `(*env)->DeleteLocalRef(env, instance)` in the cleanup of this method, so that we don't overflow the LREF table.